### PR TITLE
Awesome Books with Modular JavaScript

### PR DESCRIPTION
**Pull request by sharon-odhiambo**
**In this PR I used Modular JavaScript to build an awesome booklist where the user can add or remove a book from the booklist and  the data is stored in the local storage.**

**Highlights**

- Initialize the repo using git
- Add the booklist markup using index.html
- Create an entry point JS file
- Create JS modules that contain the different sections
- Reconfigure linters and fix linter errors.
- Add time module from Luxon library.

**Cheers and Happy Reviewing**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  e.preventDefault();
+  const title = document.querySelector('#title').value;
+  const author = document.querySelector('#author').value;
+  const book = new Book(title, author);
+  UI.addBookToList(book);
+  Store.addBook(book);
+  UI.clearFields();
+});
+document.querySelector('.book-list').addEventListener('click', (e) => {
+  UI.deleteBook(e.target);
```
Comment (omarsalem7) [inline index.js:20]: - [ ] It will be better if you add a function to delete the book from local storage like you did when you add the book.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @sharon-odhiambo 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- Descriptive pull request ✔️
- No linter errors ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/storage.js [hunk 2]
```diff
@@ -23,5 +24,15 @@ export class Store {
...
+  static removeBook(isbn) {
+    const books = Store.getBooks();
+    books.forEach((book, index) => {
+      if (book.isbn === isbn) {
+        books.splice(index, 1);
+      }
+    });
+    localStorage.setItem('books', JSON.stringify(books));
+  }
 }
```
Comment (zieeco) [inline modules/storage.js]: - As required by the previous reviewer, please ensure the deleted book is updated in the localStorage and the deleted book should be removed from the UI. As of now, when I reload your page, the deleted shows up again on your UI. Please, kindly fix this..

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @sharon-odhiambo ,


Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sharon-odhiambo** [comment]
Hi @zieeco,
Thank you for your review.
However, that is not true on my end.
Kindly check below my UI and local storage when I remove a book.
[image: Screenshot (14).png][image: Screenshot (12).png][image: Screenshot
(13).png]

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @sharon-odhiambo 

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**